raft/src/main/java/org/apache/kafka/raft/ReplicatedCounter.java

```diff
@@ -106,7 +108,10 @@ public synchronized void handleCommit(BatchReader<Integer> reader) {
                     lastCommittedEpoch,
                     lastOffsetSnapshotted
                 );
-                Optional<SnapshotWriter<Integer>> snapshot = client.createSnapshot(lastCommittedOffset, lastCommittedEpoch, 0);
+                Optional<SnapshotWriter<Integer>> snapshot = client.createSnapshot(
```

Comment:
Do we need to assert that the `BatchReader` returned at least one record? If we don't, isn't it possible we could create a snapshot with lastCommittedOffset = -1, lastCommittedTimestamp = -1, etc. ?

So I guess we could just add a check that lastCommittedOffset != -1.

Reply:
We are doing this within this check:
```
if (lastOffsetSnapshotted + snapshotDelayInRecords  < lastCommittedOffset) {
```
As long as `snapshotDelayInRecords` is greater than `0`, which is the case here, I think we can be sure that `lastCommittedOffset` is not `-1`.